CppCoreGuidelines.md

```diff
@@ -13281,6 +13281,10 @@ Helps make style consistent and conventional.
 By definition, a condition in an `if`-statement, `while`-statement, or a `for`-statement selects between `true` and `false`.
 A numeric value is compared to `0` and a pointer value to `nullptr`.
 
+    // Assuming that "my_condition" is a `bool` variable
+    if (my_condition) { ... }          // good
+    if (my_condition == true) { ... }  // redundant ==true, not recommended
```

Comment:
would have been better if you had another example:
```
if (true == my_condition) { ... }  // redundant but preferred if it must be specified
```

Reason being - some like myself prefer to have the redundancy obvious - being explicit over implicit. However, having the `true` as the l-value and the variable as the r-value prevents errors with `=` vs `==` from happening.

So there should be a middle ground between the implicit `my_condition` and the explicit.

Reply:
@BenjamenMeyer Thank you for your suggestion. I know that `if (true == my_condition)` would prevent an accidental assignment, that would happen when doing `if (my_condition = true)`, accidentally. But honestly, I still prefer to have the "unknown" (the variable) at the left side, and the "known value" (the constant) at the right side, because I find it more intuitive.